### DISCUSSION TITLE
How to find individual words and their count?

### DISCUSSION THREAD

**Discussion by kartikthapar** [BurntSushi/ripgrep#1908] [General] [answered]
Consider the search pattern `alpha` which can be found in different ways like: `alpha`, `alpha1`, `alpha2` or `1alpha`, `2alpha` and so on. I would like to run a search that gives me the following output:

| Hit | Count |
| --- | --- |
| `alpha` | 150 |
| `alpha1` | 100 |
| `1alpha` | 200 |

and so on.

What is the best way to accomplish this using ripgrep?

**Comment by BurntSushi** [comment, marked as answer]
For this very specific situation, this is what I would do:

```
$ cat corpus1
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha

$ cat corpus2
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha
alpha alpha1 alpha2 1alpha 2alpha

$ for w in alpha alpha1 alpha2 1alpha 2alpha; do count="$(rg -w "$w" -o | wc -l)"; echo "$w\t$count"; done
alpha   12
alpha1  12
alpha2  12
1alpha  12
2alpha  12
```

I suspect what you're after is a way to run ripgrep once and get output similar to this. That is not possible. ripgrep does not have any way of reporting or acting upon _which_ pattern matched.

You could do something like this too though. Kinda depends on the actual problem you're trying to solve:

```
$ rg -w '\d*alpha\d*' -o --no-filename | sort | uniq -c
     12 1alpha
     12 2alpha
     12 alpha
     12 alpha1
     12 alpha2
```
